packages/gatsby/src/state-machines/shared-transition-configs.ts

```diff
@@ -0,0 +1,35 @@
+import { TransitionConfig, AnyEventObject, DoneInvokeEvent } from "xstate"
+import { IBuildContext } from "../services"
+
+type BuildTransition = TransitionConfig<IBuildContext, AnyEventObject>
+
+/**
+ * Event handler used in all states where we're not ready to process node
+ * mutations. Instead we add it to a batch to process when we're next idle
+ */
+export const ADD_NODE_MUTATION: BuildTransition = {
+  actions: `addNodeMutation`,
+}
+
+/**
+ * Event handler used in all states where we're not ready to process a file change
+ * Instead we add it to a batch to process when we're next idle
+ */
+export const SOURCE_FILE_CHANGED: BuildTransition = {
+  actions: `markFilesDirty`,
+}
+
+/**
+ * When running queries we might add nodes (e.g from resolvers). If so we'll
+ * want to re-run queries and schema inference
+ */
+export const runMutationAndMarkDirty: BuildTransition = {
```

Comment:
Why is one constant capitalized and the other is not?

Reply:
The other one is capitalised because it's a helper to let me import it and use it directly as an event handler (it matches the name of the event)